### PR TITLE
doc: improve manager info

### PR DESCRIPTION
**Pull request by camilamacedo86**
**Description**
Provide required info about project scope definitions using namespace as an option in the manager. 

**Motivation**
https://github.com/operator-framework/operator-sdk/issues/2008

**To verify**
https://deploy-preview-1229--kubebuilder.netlify.com/cronjob-tutorial/empty-main.html

### CODE REVIEW

File: docs/book/src/cronjob-tutorial/testdata/emptymain.go [hunk 1]
```diff
@@ -97,3 +98,26 @@ func main() {
...
+```go
+mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+	Scheme: scheme,
+	Namespace: namespace,
+	MetricsBindAddress: metricsAddr}
+)
+```
+
+The above example will change the scope of your project to a Namespaced one and then, in this scenario
+would also be suggested replace the default ClusterRole and ClusterRoleBinding to Role and RoleBinding respectively.
```
Comment (Adirio) [inline docs/book/src/cronjob-tutorial/testdata/emptymain.go]:
```
The above example will change the scope of your project to a single Namespace. In this scenario,
it is also suggested to restrict the provided authorization to this namespace by replacing the default
ClusterRole and ClusterRoleBinding to Role and RoleBinding respectively
```

File: docs/book/src/cronjob-tutorial/testdata/emptymain.go [hunk 2]
```diff
@@ -97,3 +98,26 @@ func main() {
...
+For further information see the kubernetes documentation about [Using RBAC Authorization](https://kubernetes.io/docs/reference/access-authn-authz/rbac/)
+
+<aside class="note">
+<h1>Watching a list of Namespaces</h1>
+See [here](https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/cache#MultiNamespacedCacheBuilder) that for some scenarios is possible to watch a list of namespaces as well.
```
Comment (Adirio) [inline docs/book/src/cronjob-tutorial/testdata/emptymain.go]:
```
Watching a list of namespaces is possible for some scenarios as explained
[here](https://www.mdpi.com/journal/sensors/special_issues/cloud_computing_iot).
```

File: docs/book/src/cronjob-tutorial/testdata/emptymain.go [hunk 3]
```diff
@@ -97,3 +98,26 @@ func main() {
...
+</aside>
+
+ */
```
Comment (Adirio) [inline docs/book/src/cronjob-tutorial/testdata/emptymain.go]: Add a new line at the end of the file

File: docs/book/src/cronjob-tutorial/testdata/emptymain.go [hunk 4]
```diff
@@ -97,3 +98,26 @@ func main() {
...
+	MetricsBindAddress: metricsAddr}
+)
+```
+
+The above example will change the scope of your project to a Namespaced one and then, in this scenario
+would also be suggested replace the default ClusterRole and ClusterRoleBinding to Role and RoleBinding respectively.
+For further information see the kubernetes documentation about [Using RBAC Authorization](https://kubernetes.io/docs/reference/access-authn-authz/rbac/)
+
+<aside class="note">
+<h1>Watching a list of Namespaces</h1>
```
Comment (mengqiy) [inline docs/book/src/cronjob-tutorial/testdata/emptymain.go]: IMO it's unnecessary to add a note class here. And it seems markdown syntax doesn't work in this block.

File: docs/book/src/cronjob-tutorial/testdata/emptymain.go (hunk 2 above)
Comment (mengqiy) [reply docs/book/src/cronjob-tutorial/testdata/emptymain.go]: We probably should make it more clear how to do it. e.g.
```go
mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
	Scheme: scheme,
	NewCache: cache.MultiNamespacedCacheBuilder(myNamespaceList),
	MetricsBindAddress: metricsAddr,
})
```

File: docs/book/src/cronjob-tutorial/testdata/emptymain.go [hunk 5]
```diff
@@ -97,3 +98,32 @@ func main() {
...
+)
+```
+
+The above example will change the scope of your project to a single Namespace. In this scenario,
+it is also suggested to restrict the provided authorization to this namespace by replacing the default
+ClusterRole and ClusterRoleBinding to Role and RoleBinding respectively
+For further information see the kubernetes documentation about [Using RBAC Authorization](https://kubernetes.io/docs/reference/access-authn-authz/rbac/)
+
+Also, it is possible to use the [MultiNamespacedCacheBuilder](https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/cache#MultiNamespacedCacheBuilder) to watch a specific set of namespaces:
+```Go
```
Comment (mengqiy) [inline docs/book/src/cronjob-tutorial/testdata/emptymain.go]: > Go

Not sure if it matters. I may need to be `go`.

File: docs/book/src/cronjob-tutorial/testdata/emptymain.go [hunk 6]
```diff
@@ -97,3 +98,32 @@ func main() {
 		os.Exit(1)
 	}
 }
+
+/*
```
Comment (mengqiy) [inline docs/book/src/cronjob-tutorial/testdata/emptymain.go]: I'd suggest move the new contents under line `mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{Scheme: scheme, MetricsBindAddress: metricsAddr})`.

File: docs/book/src/cronjob-tutorial/testdata/emptymain.go [hunk 7]
```diff
@@ -97,3 +98,32 @@ func main() {
...
+it is also suggested to restrict the provided authorization to this namespace by replacing the default
+ClusterRole and ClusterRoleBinding to Role and RoleBinding respectively
+For further information see the kubernetes documentation about [Using RBAC Authorization](https://kubernetes.io/docs/reference/access-authn-authz/rbac/)
+
+Also, it is possible to use the [MultiNamespacedCacheBuilder](https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/cache#MultiNamespacedCacheBuilder) to watch a specific set of namespaces:
+```Go
+var namespaces []string // List of Namespaces
+
+mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+	Scheme: scheme,
```
Comment (mengqiy) [inline docs/book/src/cronjob-tutorial/testdata/emptymain.go]: In the preview, I noticed that these lines doesn't have same indentations.

File: docs/book/src/cronjob-tutorial/testdata/emptymain.go [hunk 8]
```diff
@@ -89,6 +90,40 @@ func main() {
...
+		NewCache: cache.MultiNamespacedCacheBuilder(namespaces),
+		MetricsBindAddress: fmt.Sprintf("%s:%d", metricsHost, metricsPort),
+	})
+	...
+
+	For further information see:
+	https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/cache#MultiNamespacedCacheBuilder
+	*/
+
 	// +kubebuilder:scaffold:builder
```
Comment (camilamacedo86) [inline docs/book/src/cronjob-tutorial/testdata/emptymain.go]: @mengqiy done the changes. let me know if t is ok now :-)
Really tks for the help.

File: docs/book/src/cronjob-tutorial/testdata/emptymain.go [hunk 9]
```diff
@@ -89,6 +90,40 @@ func main() {
 		os.Exit(1)
 	}
 
+	/*
+
+	Note that the Manager can restrict the namespace that all controllers will watch for resources by:
+
+	...
```
Comment (mengqiy) [inline docs/book/src/cronjob-tutorial/testdata/emptymain.go]: Use ```.

File: docs/book/src/cronjob-tutorial/testdata/emptymain.go [hunk 10]
```diff
@@ -89,6 +90,40 @@ func main() {
...
+	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		Scheme: scheme,
+		Namespace: namespace,
+		MetricsBindAddress: metricsAddr}
+	)
+	...
```
Comment (mengqiy) [inline docs/book/src/cronjob-tutorial/testdata/emptymain.go]: Same here and below.

Comment (camilamacedo86) [reply docs/book/src/cronjob-tutorial/testdata/emptymain.go]: Done 👍

File: docs/book/src/cronjob-tutorial/testdata/emptymain.go [hunk 11]
```diff
@@ -89,6 +90,40 @@ func main() {
 		os.Exit(1)
 	}
 
+	/*
+
+	Note that the Manager can restrict the namespace that all controllers will watch for resources by:
+
+	```
```
Comment (mengqiy) [inline docs/book/src/cronjob-tutorial/testdata/emptymain.go]: Use ```go

File: docs/book/src/cronjob-tutorial/testdata/emptymain.go [hunk 12]
```diff
@@ -89,6 +90,40 @@ func main() {
...
+	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		Scheme: scheme,
+		Namespace: namespace,
+		MetricsBindAddress: metricsAddr}
```
Comment (mengqiy) [inline docs/book/src/cronjob-tutorial/testdata/emptymain.go]: nit: To be consistent with the code block below, I'd put `}` to next line with `)`.
```
MetricsBindAddress: metricsAddr,
})
```

File: docs/book/src/cronjob-tutorial/testdata/emptymain.go [hunk 13]
```diff
@@ -89,6 +90,40 @@ func main() {
...
+
+	The above example will change the scope of your project to a single Namespace. In this scenario,
+	it is also suggested to restrict the provided authorization to this namespace by replacing the default
+	ClusterRole and ClusterRoleBinding to Role and RoleBinding respectively
+	For further information see the kubernetes documentation about Using RBAC Authorization:
+	https://kubernetes.io/docs/reference/access-authn-authz/rbac/.
+
+	Also, it is possible to use the MultiNamespacedCacheBuilder to watch a specific set of namespaces;
+
+	```
```
Comment (mengqiy) [inline docs/book/src/cronjob-tutorial/testdata/emptymain.go]: Same here

Comment (mengqiy) [reply docs/book/src/cronjob-tutorial/testdata/emptymain.go]:
```
	```go
	var namespaces []string // List of Namespaces
	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
	    Scheme: scheme,
	    NewCache: cache.MultiNamespacedCacheBuilder(namespaces),
	    MetricsBindAddress: fmt.Sprintf("%s:%d", metricsHost, metricsPort),
	})
	```
```

File: docs/book/src/cronjob-tutorial/testdata/emptymain.go (hunk 11 above)
Comment (mengqiy) [reply docs/book/src/cronjob-tutorial/testdata/emptymain.go]: I noticed that it doesn't look great in netlify preview, use some whitespaces should do the trick.
```
	```go
	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
	    Scheme: scheme,
	    Namespace: namespace,
	    MetricsBindAddress: metricsAddr,
	})
	```
```

Comment (camilamacedo86) [reply docs/book/src/cronjob-tutorial/testdata/emptymain.go]: How could I check it in netlify preview? Could you pass the link?

Comment (camilamacedo86) [reply docs/book/src/cronjob-tutorial/testdata/emptymain.go]: Tried to fix the spaces anyway.

Comment (mengqiy) [reply docs/book/src/cronjob-tutorial/testdata/emptymain.go]: It's in the github check list.
Here's the link: https://deploy-preview-1229--kubebuilder.netlify.com/

Comment (camilamacedo86) [reply docs/book/src/cronjob-tutorial/testdata/emptymain.go]: Really tks I did not found that.

Comment (camilamacedo86) [reply docs/book/src/cronjob-tutorial/testdata/emptymain.go]: It is definitely something that we could add in the CONTRIBUTION.MD

**Comment by camilamacedo86** [comment]
Hi @mengqiy now it is 100% OK :-)

**Pull request merged**